src/api/search/test/query.test.js

```diff
@@ -1,9 +1,229 @@
+        {
+          msg: 'Invalid value(s)',
+          param: '_error',
+          nestedErrors: [
+            { msg: 'post should not be empty', param: 'post', location: 'body' },
+            { msg: 'author should exist', param: 'author', location: 'body' },
+            { msg: 'title should exist', param: 'title', location: 'body' },
+          ],
+        },
+      ]);
+    });
+
+    it('Return error 400 if "to" param has an invalid date format', async () => {
+      let res = await request(app)
+        .get('/advanced')
+        .query({ post: 'ElasticSearch', to: 'invalid date format' });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toStrictEqual('invalid date format');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', to: '06/04/2020' });
+      expect(res.status).toBe(400);
+      expect(res.body).toStrictEqual([
+        {
+          value: '06/04/2020',
+          msg: 'invalid date format',
+          param: 'to',
+          location: 'query',
+        },
+      ]);
+    });
+
+    it('Return error 400 if "from" param has an invalid date format', async () => {
+      let res = await request(app)
+        .get('/advanced')
+        .query({ post: 'ElasticSearch', from: 'foo-bar 2020' });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toStrictEqual('foo-bar 2020');
+
+      res = await request(app)
+        .get('/advanced')
+        .query({ post: 'ElasticSearch', from: '2020-25-23' });
+      expect(res.status).toBe(400);
+      expect(res.body).toStrictEqual([
+        {
+          value: '2020-25-23',
+          msg: 'invalid date format',
+          param: 'from',
+          location: 'query',
+        },
+      ]);
+    });
+
+    it('Return error 400 if "perPage" param is not an integer between 1 to 10', async () => {
+      let res = await request(app).get('/advanced').query({ post: 'ElasticSearch', perPage: 11 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('11');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', perPage: 0 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('0');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', perPage: 1.5 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('1.5');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', perPage: 'one' });
+      expect(res.status).toBe(400);
+      expect(res.body).toStrictEqual([
+        {
+          value: 'one',
+          msg: 'perPage should be empty or a number between 1 to 10',
+          param: 'perPage',
+          location: 'query',
+        },
+      ]);
+    });
+
+    it('Return error 400 if "page" param is not an integer between 0 to 999', async () => {
+      let res = await request(app).get('/advanced').query({ post: 'ElasticSearch', perPage: 101 });
```

Comment:
I think this is an mistake. page `101` is within the range.

Reply:
Yeah you're right, its testing the wrong param. Oops